sdk/core/azure-core/src/main/java/com/azure/android/core/util/logging/ClientLogger.java

```diff
@@ -3,287 +3,112 @@
 
 package com.azure.android.core.util.logging;
 
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
-import java.util.Arrays;
-import java.util.Objects;
-
-/**
- * This is a fluent logger helper class that wraps a pluggable {@link Logger}.
- *
- * <p>This logger logs formattable messages that use {@code {}} as the placeholder. When a {@link Throwable throwable}
- * is the last argument of the format varargs and the logger is enabled for
- * {@link ClientLogger#verbose(String, Object...) verbose}, the stack trace for the throwable is logged.</p>
- *
- * <p><strong>Log level hierarchy</strong></p>
- * <ol>
- * <li>{@link ClientLogger#error(String, Object...) Error}</li>
- * <li>{@link ClientLogger#warning(String, Object...) Warning}</li>
- * <li>{@link ClientLogger#info(String, Object...) Info}</li>
- * <li>{@link ClientLogger#verbose(String, Object...) Verbose}</li>
- * </ol>
- *
- */
-public class ClientLogger {
-    private final Logger logger;
-
-    /*
-     * Indicates that log level is at verbose level.
-     */
-    private static final int VERBOSE_LEVEL = 1;
-
-    /*
-     * Indicates that log level is at information level.
-     */
-    private static final int INFORMATIONAL_LEVEL = 2;
-
-    /*
-     * Indicates that log level is at warning level.
-     */
-    private static final int WARNING_LEVEL = 3;
-
-    /*
-     * Indicates that log level is at error level.
-     */
-    private static final int ERROR_LEVEL = 4;
-
-    /*
-     * Indicates that logging is disabled.
-     */
-    private static final int DISABLED_LEVEL = 5;
-
+public interface ClientLogger {
     /**
-     * Retrieves a logger for the passed class using the {@link LoggerFactory}.
+     * Logs a message that at the {@code debug} log level.
      *
-     * @param clazz Class creating the logger.
-     */
-    public ClientLogger(Class<?> clazz) {
-        this(clazz.getName());
-    }
-
-    /**
-     * Retrieves a logger for the passed class name using the {@link LoggerFactory}.
+     * <p><strong>Code samples</strong></p>
+     *
+     * <p>Logging a message at the debug log level.</p>
+     * <p>
+     * {@codesnippet com.azure.core.util.logging.clientLogger.info}
```

Comment:
@bsiegel I brought this up the first time we reviewed the ClientLogger and we didn't really discussed it any further. I would prefer to remove all these references and add them when necessary, since no other piece of code has anything like it yet. 
Keep it simple. What do you think?

Reply:
Sounds good to me, as part of march towards GA we will make sure to add snippets & snippet references. Until then I'm fine with them either being incorrect or missing.